hooks/boot/01-kafo-hook-extensions.rb

```diff
@@ -31,9 +31,13 @@ def ensure_packages(packages, state = 'installed')
     end
   end
 
-  def apply_puppet_code(code)
-    bin_path = Kafo::PuppetCommand.search_puppet_path('puppet')
-    Open3.capture3(*Kafo::PuppetCommand.format_command("echo \"#{code}\" | #{bin_path} apply --detailed-exitcodes"))
+  def apply_puppet_code(code, use_noop=true)
```

Comment:
Feels like this whole thing would live in Kafo to be honest.

Reply:
Kind of, but I wasn't sure how things would behave if you have multiple definitions of that. I'd still be open to it.